modifytags/modifytags_test.go

```diff
@@ -0,0 +1,133 @@
+package modifytags
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var update = flag.Bool("update", false, "update golden (.out) files")
+
+// This is the directory where our test fixtures are.
+const fixtureDir = "./test-fixtures"
+
+func TestApply(t *testing.T) {
+	var tests = []struct {
+		file  string
+		m     *Modification
+		start token.Pos
+		end   token.Pos
+	}{
+		{
+			file: "all_structs",
+			m: &Modification{
+				Add:       []string{"json"},
+				Transform: SnakeCase,
+			},
+			start: token.Pos(1),
+			end:   token.NoPos, // will be set to end of file
+		},
+		{
+			file: "clear_all_tags",
+			m: &Modification{
+				Clear: true,
+			},
+			start: token.Pos(1),
+			end:   token.NoPos, // will be set to end of file
+		},
+		{
+			file: "remove_some_tags",
+			m: &Modification{
+				Remove: []string{"json"},
+			},
+			start: token.Pos(1),
+			end:   token.NoPos, // will be set to end of file
+		},
+		{
+			file: "add_tags_pos_between_line",
+			m: &Modification{
+				Add: []string{"json"},
+				AddOptions: map[string][]string{
+					"json": {"omitempty"},
+				},
+				Transform: SnakeCase,
+			},
+			start: token.Pos(50),  // middle of second struct field
```

Comment:
Maybe just leave a TODO to extract these from the source. E.g.

// TODO: use markers in the test content to delineate start and end, rather than hard-coding here.